locksmith/src/utils/certification.tsx

```diff
@@ -0,0 +1,117 @@
+import normalizer from './normalizer'
+import config from '../config/config'
+import * as metadataOperations from '../operations/metadataOperations'
+import { Certificate, minifyAddress } from '@unlock-protocol/ui'
+import { SubgraphKey } from '@unlock-protocol/unlock-js'
+import satori from 'satori'
+import dayjs from 'dayjs'
+
+import { readFileSync } from 'fs'
+import { MAX_UINT } from '../../constants'
+import { imageToBase64 } from './image'
+import { networks } from '@unlock-protocol/networks'
+import { generateKeyMetadata } from '../operations/metadataOperations'
+const inter400 = readFileSync('src/fonts/inter-400.woff')
+const inter700 = readFileSync('src/fonts/inter-700.woff')
+
+interface Options {
+  network: number
+  lockAddress: string
+  tokenId: string
+  key?: Omit<SubgraphKey, 'lock'>
+}
+
+/**
+ * This generates a certificate using the satori library.
+ */
+export const createCertificate = async ({
+  network,
+  lockAddress: lock,
+  tokenId,
+  key,
+}: Options) => {
+  const lockAddress = normalizer.ethereumAddress(lock)
+
+  const [lockData, metadata] = await Promise.all([
+    metadataOperations.getLockMetadata({
+      lockAddress,
+      network,
+    }),
+    generateKeyMetadata(
+      lockAddress,
+      tokenId,
+      true,
+      config.services.locksmith,
+      network
+    ),
+  ])
+
+  const attributes: Record<string, string>[] = lockData?.attributes || []
+
+  const object = attributes.reduce<Record<string, string>>(
+    (item, { trait_type, value }) => {
+      item[trait_type] = value as string
+      return item
+    },
+    {}
+  )
+
+  const transactionHash = key?.transactionsHash?.[0] ?? ''
+
+  const expiration =
+    key?.expiration && key?.expiration === MAX_UINT
+      ? ''
+      : dayjs.unix(key?.expiration).format('DD MMM YYYY') // example ('18 Apr 2023')
+
+  const imageBase64 = await imageToBase64(metadata?.image)
+
+  const certificate = await satori(
+    <div
+      style={{
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        height: '100%',
+        width: '100%',
+      }}
+    >
+      <Certificate
+        name={metadata?.name || ''}
+        description={metadata.description}
+        tokenId={metadata?.tokenId}
+        owner={metadata?.owner}
+        issuer={object?.certification_issuer}
+        image={imageBase64}
+        lockAddress={minifyAddress(lockAddress)}
+        network={network}
+        networkName={networks[network]?.name}
+        // can't use minifyAddress because is checking if the address is valid
+        transactionsHash={`${transactionHash.slice(
```

Comment:
Please create a util function and add to the core package?

Reply:
we have it already in `ui`:

```
export const minifyAddress = (address: string) => {
  const checked = ethers.utils.isAddress(address)
  return checked
    ? `${address.slice(0, 5)}...${address.slice(address.length - 5)}`
    : address
}
```
should not work for transaction hash because we verify that we pass an address for minify

Reply:
ok but there is no value in displaying a truncated version if there is no link, because nobody can do anything useful with a truncated hash... so either you show the full hash or you point to the block explorer.

Reply:
In this case for the generated image the link is not present, but is present on the certification detail https://github.com/unlock-protocol/unlock/blob/master/unlock-app/src/components/content/certification/CertificationDetails.tsx#L215